Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,685 @@
+                testlogApi = new QTestApi.TestlogApi(connObj.Configuration);
+                attachmentApi = new QTestApi.AttachmentApi(connObj.Configuration);
+
+                QtestTestSuite testSuite = GetQtestTestSuite(bizFlow.ExternalID);
+                if (testSuite != null)
+                {
+                    //get all BF Activities groups
+                    ObservableList<ActivitiesGroup> activGroups = bizFlow.ActivitiesGroups;                    
+                    if (activGroups.Count > 0)
+                    {
+                        foreach (ActivitiesGroup activGroup in activGroups)
+                        {
+                            if ((publishToALMConfig.FilterStatus == FilterByStatus.OnlyPassed && activGroup.RunStatus == eActivitiesGroupRunStatus.Passed)
+                            || (publishToALMConfig.FilterStatus == FilterByStatus.OnlyFailed && activGroup.RunStatus == eActivitiesGroupRunStatus.Failed)
+                            || publishToALMConfig.FilterStatus == FilterByStatus.All)
+                            {
+                                QtestTest tsTest = null;
+                                //go by TC ID = TC Instance ID
+                                tsTest = testSuite.Tests.Where(x => x.TestID == activGroup.ExternalID).FirstOrDefault();                              
+                                if (tsTest != null)
+                                {
+                                    //get activities in group
+                                    List<Activity> activities = (bizFlow.Activities.Where(x => x.ActivitiesGroupID == activGroup.Name)).Select(a => a).ToList();
+                                    string TestCaseName = PathHelper.CleanInValidPathChars(tsTest.TestName);
+                                    if ((publishToALMConfig.VariableForTCRunName == null) || (publishToALMConfig.VariableForTCRunName == string.Empty))
+                                    {
+                                        String timeStamp = DateTime.Now.ToString("dd-MMM-yyyy HH:mm:ss");
+                                        publishToALMConfig.VariableForTCRunName = "GingerRun_" + timeStamp;
+                                    }
+
+                                    //RunFactory runFactory = (RunFactory)tsTest.RunFactory;
+                                    //Run run = (Run)runFactory.AddItem(publishToALMConfig.VariableForTCRunNameCalculated);
+
+                                    if (tsTest.Runs[0] != null)
+                                    {
+                                        List<QTestApiModel.StatusResource> statuses = testrunApi.GetStatusValuable((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey));
+                                        List<QTestApiModel.StatusResource> stepsStatuses = new List<QTestApiModel.StatusResource>();
+                                        QTestApiModel.StatusResource testCaseStatus = new QTestApiModel.StatusResource();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'testCaseStatus'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)